### PR TITLE
Enhanced checking of the requirements...

### PR DESCRIPTION
**Pull request by pjaspers**
I tried rendering a gif with subtitles and ran into an error like this:

```
convert.im6: no decode delegate for this image format `/tmp/magick-dmhxdNv4' @ error/constitute.c/ReadImage/544.
convert.im6: no images defined `gif:-' @error/convert.c/ConvertImageCommand/3044.
```

This is because `convert` doesn't get any input from `ffmpeg` and this
was caused because `ffmpeg` was not compiled with the `libass` flag.

So I added some extra dependency checking in the `checkRequirements`
call.

This does seem to add roughly 0.02s to the app, based on my
extensive (read: running the app with `time`, like an animal).

The resulting gif in all its glory:

WAT

### CODE REVIEW

File: bin/gifify [hunk 1]
```diff
@@ -72,19 +74,43 @@ function encode() {
...
+function missingDependency(command, dependencies) {
+  return ansiRed(util.format("\t%s needs %s", command, dependencies.join(", ")));
+}
 
-      resolve();
+// This will check both if the required applications are installed, and if they
+// are installed with the correct dependencies. For example, to make subtitles
+// work, ffmpeg needs to be compiled with `enable-libass`.
+function checkRequirements() {
+  var required = {
```
Comment (vvo) [inline bin/gifify]: On a side note, if you had made required an array you could have do:

```js
return Promise.all(required.map(checkRequirement));
```

Shiny? @pjaspers 

I always use an array when in need of iterating. Only use a map if need to direct access a value from the map like var som = map[keyName];

File: bin/gifify [hunk 2]
```diff
@@ -72,19 +74,43 @@ function encode() {
   outputStream.on('error', abort.bind(null));
 }
 
-function checkRequirements() {
-  var requirements = ['ffmpeg', 'convert', 'gifsicle'];
-  return Promise.all(requirements.map(findRequirement));
+function ansiRed(string) {
```
Comment (vvo) [inline bin/gifify]: Can you use the colors module? Would reduce the code

**Comment by vvo** [comment]
@pjaspers This is truly awesome. Could you add a link to the readme requirements when something fails?

This + some refactoring if you are up to. Great job!!!

**Comment by pjaspers** [comment]
@vvo done and done.

**Comment by vvo** [comment]
Thanks a lot @pjaspers !! Nice job :)

**Pull request merged**